### DISCUSSION TITLE
Integration with Appynamics for Monitoring

### DISCUSSION THREAD

**Discussion by jiyer2016** [bentoml/BentoML#1059] [General] [answered]
My environment AppDynamics to monitor app.
I am still trying to figure out how to setup AppD to monnitor BentoML (both the YataiService and the Model Service End Point).
Here is the official documentation:
https://docs.appdynamics.com/display/PRO45/Install+the+Python+Agent

In the meanwhile - wanted to know if others have tried this out ?
Do you think BentoML can be monitored by AppDynamics out-of-box ?
While I see references to gunicorn in there - I can't find a mention for gRPC.

**Comment by parano** [comment, marked as answer]
Hi @jiyer2016, for Model Service endpoints, you can potentially use BentoML's Prometheus `/metrics` endpoint and report it to AppDynamics? https://www.appdynamics.com/community/exchange/prometheus-monitoring-extension/ 

BentoML's model service is not a simple WSGI app, it runs an aiohttp server that does adaptive micro-batcing in front of the model backend which runs inside a flask/gunicorn server. The standard appdynamic + WSGI app integration probably won't work out-of-the-box wit BentoML.

As for YataiService, we still don't have a good monitoring story around it. Although it is on our roadmap, we are planning to focus on improving YataiService in the next two months. I will make sure to cover the monitoring side of things(at least provide a prometheus metrics endpoint).
